src/m365/spo/commands/site/site-recyclebinitem-clear.spec.ts

```diff
@@ -0,0 +1,192 @@
+    await command.action(logger, {
+      options: {
+        siteUrl: 'https://contoso.sharepoint.com'
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the items from the recycle bin when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'post');
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+    await command.action(logger, {
+      options: {
+        siteUrl: 'https://contoso.sharepoint.com'
+      }
+    });
+
+    assert(postSpy.notCalled);
+  });
+
+  it('removes all items from the first-stage recycle bin and confirm option', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/RecycleBin/DeleteAll`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.doesNotReject(command.action(logger, {
+      options: {
+        verbose: true,
+        siteUrl: 'https://contoso.sharepoint.com',
+        confirm: true
+      }
+    }));
+  });
+
+  it('removes all items from the first-stage recycle bin without confirmation', async () => {
+    const postSpy = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/RecycleBin/DeleteAll`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, {
+      options: {
+        siteUrl: 'https://contoso.sharepoint.com'
+      }
+    });
+
+    assert(postSpy.called);
+  });
+
+  it('removes all items from the second-stage recycle bin with confirm option', async () => {
+    const postSpy = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/site/RecycleBin/DeleteAllSecondStageItems`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        verbose: true,
+        siteUrl: 'https://contoso.sharepoint.com',
+        secondary: true,
+        confirm: true
+      }
+    });
+
+    assert(postSpy.called);
+  });
+
+  it('handles error correctly', async () => {
+    const error = {
+      error: {
+        message: "The files cannot be moved to the second-stage recycle bin"
+      }
+    };
+
+    sinon.stub(request, 'post').callsFake(async () => {
+      throw error;
+    });
+
+    await assert.rejects(command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com', confirm: true } } as any), error.error);
```

Comment:
same here, can't wrap it in a `CommandError` object

Reply:
This is because your error message is not well structured. It should look to something like this:

```typescript
const error = {
      error: {
        'odata.error': {
          message: {
            value: "The files cannot be removed from the second-stage recycle bin."
          }
        }
      }
    };
```

Now the test actually fails when the error message doesn't match. Before this, the test always succeeded, no matter what you entered as a value. This is not really a useful test. Let's try to avoid this.